tests/test_spharm.py

```diff
@@ -0,0 +1,286 @@
+"""Tests of invariants for spharm package.
+
+Mostly the tests of a single function check that the outputs are the
+right shape and neither nan nor infinity (and obtained without
+crashes).  Occasionally there are checks of other properties.
+
+There are a few checks that we can round-trip information.
+
+"""
+
+import itertools
+
+import numpy as np
+import pytest
+
+import spharm
+
+# T799 transforms result in nans
+# T382 on a 512x1024 grid seems to work fine
+NTRUNC = [21, 42, 63]  # , 85, 106, 159]#, 255, 382]
+NLATS = [32, 64, 96]  # , 128, 160, 256]#, 384, 512]
```

Comment:
The running with three members in each list finishes in about a minute and a half.  That's probably good enough for quick checks of whether something broke by accident.

Running with eight in each list takes about half an hour.  That time probably means that sort of run should be reserved to check major changes to the Fortran code or the like.

I could probably cut that down some by skipping tests with `21 < ntrunc < NLATS // 4`, or by choosing `ntrunc` from the values that would *just* avoid aliasing of linear, quadratic, or cubic terms on the given grid.